File: Factory/C#/Logistics/Factory/Transport.cs
using System.Text;

namespace Factory
{
    public class Transport
    {
        private readonly string _type;
        private readonly string _whatToDeliver;
        private readonly Coordinate[] _transportPoints;

        public Transport(string type, string what, Coordinate[] transportPoints)
        {
            _type = type;
            _whatToDeliver = what;
            _transportPoints = transportPoints;
        }

        public string DoTransport()
        {
            var builder = new StringBuilder("Transporting by boat through points\n");

            foreach (var transportPoint in _transportPoints)
            {
                builder.Append(transportPoint);
            }

            return builder.ToString();

        }

        public string Deliver()
        {
            return $"Delivering {_whatToDeliver}s by Boat";
        }
    }
}

File: Factory/Java/LogisticsFactory/src/LogisticsFactory/Transport.java
package LogisticsFactory;

public abstract class Transport {
	protected String _type;
	protected String _whatToDeliver;
	protected Coordinate[] _transportPoints;
	
	public Transport(String type, String what, Coordinate[] transportPoints) {
		_type = type;
		_whatToDeliver = what;
		_transportPoints = transportPoints;
	}

	public String transport(){
		if(_transportPoints == null)
			return "No points to transport on";
		
		StringBuilder builder = new StringBuilder(String.format("Transporting by %s through points\n", _type));
		for(Coordinate c : _transportPoints) {
			builder.append(c.toString());
			builder.append(" ");
		}
		
		return builder.toString();
	}
	
	public String deliver()
	{
		return String.format("Delivering %s by %s",_whatToDeliver, _type);
	}
}
